src/main/java/catan/services/impl/PlayServiceImpl.java

```diff
@@ -211,37 +210,32 @@ private void buyCard(GameUserBean gameUser, GameBean game, Resources usersResour
+        cardUtil.validateUserDidNotUsedCardsInCurrentTurn(game);
+        cardUtil.validateUserDidNotBoughtCardInCurrentTurn(gameUser, DevelopmentCard.YEAR_OF_PLENTY);
 
         HexType firstResource = validateResourceType(firstResourceString);
         HexType secondResource = validateResourceType(secondResourceString);
 
-        Resources userResources = gameUser.getResources();
-        int currentFirstResourceQuantity = userResources.quantityOf(firstResource);
-        userResources.updateResourceQuantity(firstResource, currentFirstResourceQuantity + 1);
-        int currentSecondResourceQuantity = userResources.quantityOf(secondResource);
-        userResources.updateResourceQuantity(secondResource, currentSecondResourceQuantity + 1);
-
+        cardUtil.giveTwoResourcesToPlayer(gameUser, firstResource, secondResource);
         log.debug("Player got resources: {}, {}", firstResource, secondResource);
 
-        gameUser.getDevelopmentCards().decreaseQuantityByOne(DevelopmentCard.YEAR_OF_PLENTY);
-        gameUser.getDevelopmentCardsReadyForUsing().decreaseQuantityByOne(DevelopmentCard.YEAR_OF_PLENTY);
+        cardUtil.takeDevelopmentCardFromPlayer(gameUser, DevelopmentCard.YEAR_OF_PLENTY);
         game.setDevelopmentCardUsed(true);
     }
 
-    private void validateUserDidNotBoughtCardInCurrentTurn(GameUserBean gameUser, DevelopmentCard developmentCard) throws PlayException {
-        if (gameUser.getDevelopmentCardsReadyForUsing().quantityOf(developmentCard) == 0) {
-            log.debug("Cannot use card. It was bought in current turn: {}", developmentCard);
-            throw new PlayException(CARD_BOUGHT_IN_CURRENT_TURN_ERROR);
-        }
-    }
+    private void useCardMonopoly(final GameUserBean gameUser, GameBean game, String resourceString, Map<String, String> returnedParams) throws PlayException, GameException {
+        cardUtil.validateUserDidNotUsedCardsInCurrentTurn(game);
+        cardUtil.validateUserDidNotBoughtCardInCurrentTurn(gameUser, DevelopmentCard.MONOPOLY);
 
-    private void validateUserDidNotUsedCardsInCurrentTurn(GameBean game) throws PlayException {
-        if (game.isDevelopmentCardUsed()) {
-            log.debug("Cannot use card. It was already used in current turn");
-            throw new PlayException(CARD_ALREADY_USED_IN_CURRENT_TURN_ERROR);
-        }
+        HexType resource = validateResourceType(resourceString);
+
+        Integer takenResourcesCount = cardUtil.takeResourceFromRivalsAndGiveItAwayToPlayer(gameUser, game, resource);
+        log.debug("Player got {} of {} resources", takenResourcesCount, resource);
+
+        returnedParams.put("resourcesCount", takenResourcesCount.toString());
+
+        cardUtil.takeDevelopmentCardFromPlayer(gameUser, DevelopmentCard.MONOPOLY);
+        game.setDevelopmentCardUsed(true);
     }
 
     private HexType validateResourceType(String resourceString) throws PlayException {
```

Comment:
it is not clear that method returns something, rename it to "toValidResourceType":
'to..ResourceType' - says that it will convert input parameter to something and return it
'...Valid...' - says that it will validate input parameter